src/Dapr.AspNetCore/DaprEndpointRouteBuilderExtensions.cs

```diff
@@ -33,14 +36,27 @@ public static IEndpointConventionBuilder MapSubscribeHandler(this IEndpointRoute
             {
                 var dataSource = context.RequestServices.GetRequiredService<EndpointDataSource>();
                 var entries = dataSource.Endpoints
-                    .Select(e => e.Metadata.GetMetadata<TopicAttribute>()?.Name)
-                    .Where(n => n != null)
+                    .OfType<RouteEndpoint>()
+                    .Where(e => e.Metadata.GetMetadata<TopicAttribute>()?.Name != null)   // only endpoints which have  TopicAttribute with not null Name.
+                    .Where(e => e.RoutePattern.Parameters.Count == 0) // only endpoints which don't have parameters.
                     .Distinct()
-                    .ToArray();
+                    .Select(e => (e.Metadata.GetMetadata<TopicAttribute>().Name, e.RoutePattern.RawText));
```

Comment:
You might also think about reporting an error if an endpoint has a parameterful route as well. 

Right now the experience would be that it would silently fail.

Reply:
@ryanvolum  Do you mean return error when Dapr calls to get subscribed topics, or just log it as an error and return rest of the topics? How to get logger which has been configured by user, context.RequestServices.GetService<ILogger>() retruns null.

Reply:
Added a warning in this case.
I have a question, I am getting logger as
`context.RequestServices.GetService<ILogger<TopicAttribute>>()`, Is it advisable to use a new class (TopicSubscription) instead of TopicAttribute to get logger so that log is more clear.